### PR TITLE
Doctrine ODM MongoDB Mapping - rebased and cleaned up

### PR DESCRIPTION
**Pull request by prolic**
Taken some parts from @fabiocarneiro and fixed embedded mapping for mongodb.

Ping @danizord: Good to merge?

### CODE REVIEW

**Comment by prolic** [comment]
PHP7 build fails because of missing mongo-db extension. Perhabs you should remove the PHP7 build?

**Comment by fabiocarneiro** [comment]
<3

**Comment by prolic** [comment]
Perhabs we should not provide .dist configs and instead configure everything already. What do you think?

**Comment by danizord** [comment]
@prolic Yep, I think so.

File: src/ODM/MongoDB/Types/CurrencyType.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    public function convertToPHPValue($value)
+    {
+        if ($value === null || $value instanceof Currency) {
+            return $value;
+        }
+
+        $val = new Currency($value);
+
+        if (!$val) {
```
Comment (danizord) [inline src/ODM/MongoDB/Types/CurrencyType.php]: Should probably check if `$value` is string before intantiation, then `return new Currency($value);`.

File: .travis.yml [hunk 2]
```diff
@@ -5,10 +5,11 @@ php:
     - 5.6
     - 7
```
Comment (danizord) [inline .travis.yml]: Remove this

File: src/ODM/MongoDB/Types/CurrencyType.php (hunk 1 above)
Comment (prolic) [inline src/ODM/MongoDB/Types/CurrencyType.php]: The Currency's constructor checks already for invalid values.

Comment (danizord) [inline src/ODM/MongoDB/Types/CurrencyType.php]: So you can remove this check, this if will never pass

File: config/module.config.php [hunk 3]
```diff
@@ -1,7 +1,8 @@
 <?php
 
+namespace ZFBrasil\DoctrineMoneyModule;
```
Comment (danizord) [inline config/module.config.php]: Revert this change

**Comment by prolic** [comment]
done.

**Comment by danizord** [comment]
@prolic thanks!
